src/main.rs

```diff
@@ -283,7 +283,10 @@ fn main() {
         let working_dir = tmp_path.join(format!("{}", i));
         let source_root = source_root.clone();
 
-        let t = thread::spawn(move || {
+        let thread_name = String::from("");
+        thread_name.push(i.to_string());
```

Comment:
The name should be meaningful, only assigning an integer will not improve the situation much. We should use a name like `producer-` + i.

Reply:
Ok! I was thinking about this too, I'll add it now. But I'm not sure this code compiles either with my modifications, once I get the program running I'll correct this.